### PR TITLE
feat(MeetingsJsonAdapter): add settings control

### PR DESCRIPTION
**Pull request by karinasigartau0798**
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-235304

### CODE REVIEW

File: src/adapters/MeetingsJSONAdapter.js [hunk 1]
```diff
@@ -56,7 +57,8 @@ export const ROSTER_CONTROL = 'member-roster';
  *     "remoteAudio": {},
  *     "remoteVideo": {},
  *     "remoteShare": {},
- *     "state": "JOINED"
+ *     "state": "JOINED",
+ *     "showSettings": false
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:61]: this needs to be documented in the interfaces as well, can you please create a task?

Comment (cipak) [reply src/adapters/MeetingsJSONAdapter.js:61]: nvm, I added the subtask https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-236053

File: src/adapters/MeetingsJSONAdapter.js [hunk 2]
```diff
@@ -188,6 +197,7 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
           remoteShare: null,
           state: null,
           showRoster: null,
+          showSettings: null,
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:200]: maybe false would be a better value

File: src/adapters/MeetingsJSONAdapter.js [hunk 3]
```diff
@@ -743,4 +755,61 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
...
+      tooltip: 'Show settings panel',
+      state: MeetingControlState.INACTIVE,
+      text: 'Settings',
+    };
+
+    const initialState$ = new Observable((observer) => {
+      const meeting = this.datasource[ID];
+
+      if (meeting) {
+        const initialControl = meeting.showSettings ? active : inactive;
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:799]: pls use initialState instead of initialControl

File: src/data/meetings.js [hunk 4]
```diff
@@ -60,6 +64,7 @@ export default {
     remoteShare: null,
     state: MeetingState.JOINED,
     showRoster: true,
+    showSettings: true,
```
Comment (cipak) [inline src/data/meetings.js:67]: is this changing the UI somewhere?

Comment (karinasigartau0798) [reply src/data/meetings.js:67]: Yes, this change is visible in the storybook. In this case, in the corresponding story in WebexMeeting, the settings control will be initially in the ACTIVE state and will be red.

File: src/adapters/MeetingsJSONAdapter.js (hunk 1 above)
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:61]: please add it before state

File: src/adapters/MeetingsJSONAdapter.js [hunk 5]
```diff
@@ -743,4 +755,61 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
       observer.complete();
     });
   }
+
+  /**
+   * Handles the toggle of the settings.
+   *
+   * @param {string} ID ID of the meeting for which to update display
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:762]: please add a double space between the arg name and arg description. Also, description to be more specific: "Id of the meeting for which to toggle the settings flag" (or similar)

File: src/adapters/MeetingsJSONAdapter.js [hunk 6]
```diff
@@ -743,4 +755,61 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
...
+
+    meeting.showSettings = !meeting.showSettings;
+    document.dispatchEvent(new CustomEvent(SETTINGS_CONTROL, {detail: meeting}));
+  }
+
+  /**
+   * Returns an observable that emits the display data of a settings control.
+   *
+   * @private
+   * @param {string} ID ID of the meeting to toggle settings
```
Comment (cipak) [inline src/adapters/MeetingsJSONAdapter.js:776]: double space

**Review by cipak — APPROVED** [review]

**Pull request merged**